src/Control/Middleware/TrustedProxyMiddleware.php

```diff
@@ -145,8 +145,15 @@ public function process(HTTPRequest $request, callable $delegate)
             // Replace host
             foreach ($this->getProxyHostHeaders() as $header) {
                 $hostList = $request->getHeader($header);
-                if ($hostList) {
-                    $request->addHeader('Host', strtok($hostList, ','));
+
+                // Get first host (if passed through multiple proxies)
+                $host = $hostList ? strtok($hostList, ',') : null;
+
+                // Ensure the host name is valid
+                $host = filter_var($host, FILTER_VALIDATE_DOMAIN, ['flags' => FILTER_FLAG_HOSTNAME]);
```

Comment:
These two constants were added in PHP 7.0, so won't work in 5.6

Reply:
Actually, we've waited so long with this PR that we can now require PHP 7 in the `4` branch :D I've rebased and removed Max PHP 5 work again